src/zenml/integrations/mlflow/steps/mlflow_deployer.py

```diff
@@ -102,7 +102,7 @@ def mlflow_model_deployer_step(
     step_env = cast(StepEnvironment, Environment()[STEP_ENVIRONMENT_NAME])
     pipeline_name = step_env.pipeline_name
     run_name = step_env.run_name
-    step_name = step_env.step_name
+    step_name = step_env.step_run_info.pipeline_step_name
```

Comment:
@schustmi I think this change was probably too zealous. It'd be a breaking change for anyone using our `mlflow_deployer`, and in any case we don't need to change the name here, right?

Reply:
I'm guessing because it won't find the previous services deployed with the same step?

Was the reason for this name change to make it more consistent/easier to reach with the post execution workflow?

Reply:
Yes. That was [the ticket](https://zenml.atlassian.net/jira/software/projects/FRW/boards/5?assignee=61710083892c4200720ca56d&selectedIssue=FRW-1654) which more or less made sense to me, though tbh I'm not sure which of those values make for a better step name for the post-execution workflow. Your argument this morning that you'd want it to be the param name value seemed to make sense, but from what I understood of the ticket the need was for this value to be the one used in post-execution as well.

@stefannica maybe you can clarify? I don't have a strong opinion one way or another which of the two should be used.

Reply:
This is definitely what's needed to fix the bug: the previous step name is worthless, because it doesn't give me any lineage information I can use in the post-execution workflow. This new step name value is what should be "stamped" on the served model.

I would even go one step further and change `StepEnvironment.step_name` to be set to `step_run_info.pipeline_step_name` instead of what's currently being set to. The Seldon and KServe model deployers suffer from the same problem, and instead of fixing them individually, you could just correct the `step_name` value reported by the step environment. The justification is the same: the previous `StepEnvironment.step_name` value is worthless, because I can't use it anywhere in the post-execution workflow.